### PR TITLE
OGIP 26: Provide functionality to cancel tasks from tasktemplatefolder

### LINKED ISSUE #4179: Workflow Guards bei Aufgaben aus Standardabläufen
> Der Zustand «Abgebrochen» steht Teilaufgaben von sequenziellen Standardabläufen nicht zur Verfügung; Teilaufgaben werden entweder abgelehnt oder erledigt. Wenn das Abbrechen möglich wäre, würden Benutzer u.U. erwarten, dass der gesamte Standardablauf abgebrochen wird.

> Umbenannter Zustand «Abgebrochen» auf Hauptaufgaben 
> - Wird mitten im Ablauf festgestellt, dass es keinen Sinn macht, den Ablauf bis zum Ende ab-zuarbeiten, so kann der Auftraggeber den Ablauf (Hauptaufgabe) abbrechen und bricht da-mit alle Teilaufgaben ab.
> -  Dabei wird die Hauptaufgabe sowie alle noch nicht abgeschlossenen Unteraufgaben («Ge-plant», «Offen», «Abgelehnt») in den Zustand «Abgebrochen» versetzt.

Todos:
 - [x] stornieren (bzw. abbrechen siehe #4178) bei Teilaufgaben aus Standardabläufen verhindern. Das automatisierte stornieren via Hauptaufgabe soll aber nach wie vor möglich sein!
 - [x] Abbrechen von Aufgaben im Status `In Arbeit` erlauben
 - [x] Beim Abbrechen einer Hauptaufgabe müssen alle Unteraufgaben in den Status abgebrochen versetzt werden (auch mandantenübegreifende Teilaufgaben beachten).

### PR DESCRIPTION
**Pull request by phgross**
Depends on #4290

 - Add new cancel transition `in-progress` to `cancelled`, which is only available for maintask of task processes (generated from tasktemplatefolders).
 - Disallow any cancelling of the subtasks of processes. 
 - Automatically cancel all subtasks when a main_task gets cancelled

Closes #4179

### CODE REVIEW

File: opengever/task/handlers.py [hunk 1]
```diff
@@ -69,3 +70,28 @@ def reassign_team_tasks(task, event):
             'responsible',
             _(u"label_responsible", default=u"Responsible"),
             old_responsible, ITask(task).responsible)
+
+
+def cancel_subtasks(task, event):
+    if event.action not in ['task-transition-in-progress-cancelled',
```
Comment (deiferni) [inline opengever/task/handlers.py]: could we move this functionality to the task class? this should be something that you tell the object to do.

File: opengever/task/task.py [hunk 2]
```diff
@@ -320,6 +320,12 @@ def is_from_tasktemplate(self):
         """
         return IFromTasktemplateGenerated.providedBy(self)
 
+    @property
+    def is_main_task(self):
```
Comment (lukasgraf) [inline opengever/task/task.py]: This is not being used (in `progress_to_cancelled_guard()` above), and the docstring is wrong.

File: opengever/task/browser/transitioncontroller.py [hunk 3]
```diff
@@ -265,6 +264,10 @@ def uni_progress_to_closed_action(self, transition, c):
     def open_to_cancelled_guard(self, c, include_agency):
         """Checks if:
         - The current user is the issuer."""
+
```
Comment (lukasgraf) [inline opengever/task/browser/transitioncontroller.py]: Docstring needs updating (especially regarding the *why*, not so much the *what*).

**Comment by phgross** [comment]
@lukasgraf @deiferni updated according to your feedback ...

**Pull request merged**